app/helpers/application_helper/button/vm_instance_scan.rb

```diff
@@ -3,14 +3,12 @@ class ApplicationHelper::Button::VmInstanceScan < ApplicationHelper::Button::Bas
 
   def skip?
     return false if @display == "instances"
-    !(@record.is_available?(:smartstate_analysis) ||
-      @record.is_available_now_error_message(:smartstate_analysis)) ||
-    !@record.has_proxy?
+    @record.archived? || !@record.has_proxy?
   end
 
   def disable?
-    if !@record.is_available?(:smartstate_analysis)
-      return @record.is_available_now_error_message(:smartstate_analysis)
+    if !@record.supports_smartstate_analysis?
+      return @record.unsupported_reason(:smartstate_analysis)
```

Comment:
As @durandom [mentioned](https://github.com/ManageIQ/manageiq/pull/9941#discussion_r71668245) in other PR, the `disable?` method should only return boolean, so this `return` is not on correct place, but I would leave it for now, as the change is out of scope of this PR, and I'm already fixing it in my other PR: #9964. I don't have any other comments.